### PR TITLE
feat(prql): add prql support

### LINKED ISSUE #6050: PRQL support
Support for the [PRQL](https://prql-lang.org/) language would be great!

There is an issue for a tree-sitter grammar in the PRQL repo:
https://github.com/PRQL/prql/issues/1882

with an associated start of an implementation over here:
https://github.com/matthias-Q/tree-sitter-prql

### PR DESCRIPTION
**Pull request by matthias-Q**
Adds support for PRQL languange. 

I hope I added everything thats necessary to add language support. 

Closes #6050

### CODE REVIEW

**Comment by matthias-Q** [comment]
Sorry, I have not run `cargo xtasks docgen` I will do that and update the PR

**Comment by erasin** [comment]
> The only difference is that prql should be a few lines lower. should these changes be pushed?

Since `book/src/generated` is an automatically generated file, use `cargo xtasks docgen` to handle it to avoid conflicts.

I think a normal commit will do the trick. 

Alternatively Git provides some rebase and commit merge methods, so you can use `git commit --amend` to merge some changes directly into the last commit, and then push with `--force` to overwrite remote branch.

File: languages.toml [hunk 1]
```diff
@@ -2208,6 +2208,19 @@ indent = { tab-width = 2, unit = "  " }
 name = "yuck"
 source = { git = "https://github.com/Philipp-M/tree-sitter-yuck", rev = "9e97da5773f82123a8c8cccf8f7e795d140ed7d1" }
 
+[[language]]
+name = "prql"
+scope = "source.yprql"
```
Comment (the-mikedavis) [inline languages.toml:2213]: Is this a typo or is the "y" intentional?

File: runtime/queries/prql/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  (keyword_remove)
+  (keyword_intersect)
+] @keyword
+
+(literal) @string
+
+(assignment
+  alias: (field) @field)
+
+alias: (identifier) @field
```
Comment (the-mikedavis) [inline runtime/queries/prql/highlights.scm:33]:
```suggestion
  alias: (field) @variable.other.member)

alias: (identifier) @variable.other.member
```

File: runtime/queries/prql/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+(literal
+  (integer) @number)
+
+(integer) @number
+
+(literal
+  (decimal_number) @float)
+
+(decimal_number) @float
```
Comment (the-mikedavis) [inline runtime/queries/prql/highlights.scm:82]:
```suggestion
(literal
  (integer) @constant.numeric.integer)

(integer) @constant.numeric.integer

(literal
  (decimal_number) @constant.numeric.float)

(decimal_number) @constant.numeric.float
```

We use slightly different scopes than nvim. There are some below that should be switched to the Helix scopes like `@method`, `@boolean` and `@parameter`
https://docs.helix-editor.com/master/themes.html#syntax-highlighting

File: runtime/queries/prql/injections.scm [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+ (s_string) @sql
+ (#offset! @sql 0 2 0 -1)
+)
+
+(from_text
+  (keyword_from_text)
+  (keyword_json)
+  (literal) @json
+  (#offset! @json 0 3 0 -3)
+)
```
Comment (the-mikedavis) [inline runtime/queries/prql/injections.scm:11]:
```suggestion
((s_string) @injection.content
 (#set! injection.language "sql"))

(from_text
  (keyword_from_text)
  (keyword_json)
  (literal) @injection.content
  (#set! injection.language "json"))
```

We use a slightly different syntax for injections as well: https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection

File: languages.toml (hunk 1 above)
Comment (matthias-Q) [reply languages.toml:2213]: Typo, will fix that later

File: runtime/queries/prql/highlights.scm (hunk 2 above)
Comment (matthias-Q) [reply runtime/queries/prql/highlights.scm]: I guess in line 31, the last bracket should still close after `other.member` right?

File: runtime/queries/prql/highlights.scm (hunk 3 above)
Comment (matthias-Q) [reply runtime/queries/prql/highlights.scm]: Thanks, I tried to update them. I hope I catched everything

File: runtime/queries/prql/highlights.scm (hunk 2 above)
Comment (the-mikedavis) [reply runtime/queries/prql/highlights.scm]: Yep right, I deleted too much in this suggestion 😅

File: runtime/queries/prql/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+(f_string) @string.special
+(s_string) @string.special
+
+(comment) @comment
+
+(keyword_func) @keyword.function
+
+(function_call
+  (identifier) @function.call)
```
Comment (the-mikedavis) [inline runtime/queries/prql/highlights.scm:39]:
```suggestion
  (identifier) @function)
```

File: runtime/queries/prql/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  (keyword_min)
+  (keyword_max)
+  (keyword_count)
+  (keyword_count_distinct)
+  (keyword_average)
+  (keyword_avg)
+  (keyword_sum)
+  (keyword_stddev)
+  (keyword_count)
+] @function.call
```
Comment (the-mikedavis) [inline runtime/queries/prql/highlights.scm:90]:
```suggestion
] @function
```

**Review by the-mikedavis — COMMENTED** [review]
Otherwise this looks good

(Also see https://github.com/helix-editor/helix/pull/4892 which talks about separating the highlights for function definitions and references)

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks!

**Pull request merged**